perfkitbenchmarker/publisher.py

```diff
@@ -776,7 +776,7 @@ class SampleCollector(object):
       the run directory to the publishers list.
     run_uri: A unique tag for the run.
   """
-  def __init__(self, metadata_providers=None, publishers=None,
+  def __init__(self, metadata_providers=None, publishers=[],
```

Comment:
I'd prefer checking if publishers is none below and initializing it to an empty list - I like staying away from mutable default args.

Also, I notice that publishers (and not self.publishers) is getting extended online 790 and 792 - is that what's intended?